libc/test/src/sys/prctl/linux/prctl_test.cpp

```diff
@@ -0,0 +1,40 @@
+//===-- Unittests for prctl -----------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/errno/libc_errno.h"
+#include "src/sys/prctl/prctl.h"
+#include "test/UnitTest/ErrnoSetterMatcher.h"
+#include <errno.h>
+#include <sys/prctl.h>
```

Comment:
@michaelrj-google is this a UAPI header from the kernel? Guessing this is where the flags like PR_CAP_AMBIENT would be defined?

Is this the correct way for us to get those flags, or should we be providing them somehow?

Reply:
The header is defined at:

libc/include/sys/prctl.h.def

Reply:
But there is a TODO inside it.

Reply:
I'm curious about the TODO that contains. Perhaps it was alluding to a PR like this to implement them?

Reply:
From what I can tell, the prctl macros are defined in the `linux/prctl.h` header, which we include in `sys/prctl.h` but don't provide. We generally avoid defining platform specific macros, since they aren't stable, so I'd say it's fine to remove the TODO. The macros should come in through `linux/prctl.h`.

Reply:
also to clarify: It's fine to include a public header for a libc function if it's:

a) a header we provide, and

b) a fullbuild-only function.

This is necessary for things like public macros and public types.

Reply:
Some how I think we should not put `<linux/prctl.h>` inside `.def`. Maybe one could follow: https://github.com/llvm/llvm-project/blob/c3a9c905fbc486add75e16218fe58a04b7b6c282/libc/include/CMakeLists.txt#L191

Reply:
ok, thanks for the clarification @michaelrj-google . In that case, @SchrodingerZhu can you please remove the TODO in libc/include/sys/prctl.h.def ?

I don't think that prctl.h exists on Darwin and Windows. Unless we plan to support BSDs I don't see the point in making this generic based on LIBC_TARGET_OS.